source/hwIo.py

```diff
@@ -312,6 +312,16 @@ def setFeature(self, report):
 				log.debug("Set feature failed: %s" % ctypes.WinError())
 			raise ctypes.WinError()
 
+	def setOutputReport(self,data):
+		"""
+		Write the given data to the device using HidD_SetOutputReport.
+		This is instead of using the standard WriteFile which may freeze with some USB HID implementations.
+		@ param data: the report string (starting with the report ID).
+		"""
+		buf=ctypes.create_string_buffer(self._writeSize)
+		buf.raw=data
+		ctypes.windll.hid.HidD_SetOutputReport(self._writeFile,buf,self._writeSize)

```

Comment:
Could you please bring this in line with setFeature? This includes:

1. Add some debug logging
2. Execute HidD_SetOutputReport inside an if clause